### ISSUE TITLE
`torch.overrides.get_testing_overrides` does not function as intended for native tensor methods/operations

### ISSUE THREAD

**Issue by coreylammie** [pytorch/pytorch#81323]
### 🐛 Describe the bug

`torch.overrides.get_testing_overrides` (introduced in #33791) does not appear function as intended for native tensor methods/operations. For example, the following code can be used to discern the signature of `torch.Tensor.size`, as simply using `inspect.signature(torch.Tensor.size)` will yield an error:

```
import torch
import inspect

d = torch.overrides.get_testing_overrides()
print(inspect.signature(d[torch.Tensor.size])
```

This returns `size (self)`. As can be seen here: https://pytorch.org/docs/stable/generated/torch.Tensor.size.html, the expected output is `size (self, dim)`. Surprisingly, for `torch.Tensor.permute`, `permute (self, dim)` is returned. This is neither consistent with https://pytorch.org/docs/stable/generated/torch.Tensor.permute.html#torch.Tensor.permute or https://pytorch.org/docs/stable/generated/torch.permute.html#torch.permute.

### Versions

Collecting environment information...
PyTorch version: 1.11.0
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Red Hat Enterprise Linux 8.5 (Ootpa) (x86_64)
GCC version: (GCC) 8.5.0 20210514 (Red Hat 8.5.0-4)
Clang version: Could not collect
CMake version: version 3.19.6
Libc version: glibc-2.28

Python version: 3.9.12 (main, Apr  5 2022, 06:56:58)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-4.18.0-348.23.1.el8_5.x86_64-x86_64-with-glibc2.28
Is CUDA available: True
CUDA runtime version: 11.4.100
GPU models and configuration: 
GPU 0: NVIDIA GeForce GTX 1080 Ti
GPU 1: NVIDIA GeForce GTX 1080 Ti

Nvidia driver version: 470.57.02
cuDNN version: Probably one of the following:
/usr/lib64/libcudnn.so.8.1.1
/usr/lib64/libcudnn_adv_infer.so.8.1.1
/usr/lib64/libcudnn_adv_train.so.8.1.1
/usr/lib64/libcudnn_cnn_infer.so.8.1.1
/usr/lib64/libcudnn_cnn_train.so.8.1.1
/usr/lib64/libcudnn_ops_infer.so.8.1.1
/usr/lib64/libcudnn_ops_train.so.8.1.1
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] memtorch==1.1.6
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.21.5
[pip3] numpydoc==1.2
[pip3] torch==1.11.0
[pip3] torchaudio==0.11.0
[pip3] torchvision==0.12.0
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               11.3.1               h2bc3f7f_2  
[conda] ffmpeg                    4.3                  hf484d3e_0    pytorch
[conda] memtorch                  1.1.6                    pypi_0    pypi
[conda] mkl                       2021.4.0           h06a4308_640  
[conda] mkl-devel                 2018.0.1             h470d631_5  
[conda] mkl-include               2021.4.0              intel_640    intel
[conda] mkl-service               2.4.0            py39h7f8727e_0  
[conda] mkl-static                2021.4.0              intel_640    intel
[conda] mkl_fft                   1.3.1            py39hd3c417c_0  
[conda] mkl_random                1.2.2            py39h51133e4_0  
[conda] numpy                     1.21.5           py39he7a7128_1  
[conda] numpy-base                1.21.5           py39hf524024_1  
[conda] numpydoc                  1.2                pyhd3eb1b0_0  
[conda] pytorch                   1.11.0          py3.9_cuda11.3_cudnn8.2.0_0    pytorch
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] torchaudio                0.11.0               py39_cu113    pytorch
[conda] torchvision               0.12.0               py39_cu113    pytorch

**Comment by ezyang** [comment]
We're not checking for inspect consistency. I'd be happy to accept patches to fix this.
